base/compiler/abstractinterpretation.jl

```diff
@@ -3603,17 +3643,20 @@ function conditional_change(𝕃ᵢ::AbstractLattice, currstate::VarTable, condt
         newtyp = tmerge(𝕃ᵢ, newtyp, LimitedAccuracy(Bottom, oldtyp.causes))
     end
     # if this `Conditional` is from from `@isdefined condt.slot`, refine its `undef` information
-    newundef = condt.isdefined ? !then_or_else : vtype.undef
-    return StateUpdate(SlotNumber(condt.slot), VarState(newtyp, newundef), #=conditional=#true)
+    newundef = condt.isdefined ? (then_or_else === :else) : vtype.undef
+    return StateRefinement(condt.slot, newtyp, newundef)
 end
 
 function condition_object_change(currstate::VarTable, condt::Conditional,
-                                 condslot::SlotNumber, then_or_else::Bool)
+                                 condslot::SlotNumber, then_or_else::Symbol)
     vtype = currstate[slot_id(condslot)]
-    newcondt = Conditional(condt.slot,
-        then_or_else ? condt.thentype : Union{},
-        then_or_else ? Union{} : condt.elsetype)
-    return StateUpdate(condslot, VarState(newcondt, vtype.undef))
+    (thentype, elsetype) = if then_or_else === :then
+        (condt.thentype, Union{})
+    elseif then_or_else === :else
+        (Union{}, condt.elsetype)
+    else @assert false end
```

Comment:
```suggestion
    if then_or_else === :then
        thentype = condt.thentype
        elsetype = Union{}
    elseif then_or_else === :else
        thentype = Union{}
        elsetype = condt.elsetype
    else @assert false end
```
the previous version creates `Tuple` objects whose types aren't known.